### DISCUSSION TITLE
limit of payload we can send using react-native-firebase/analytics

### DISCUSSION THREAD

**Discussion by whoami-shubham** [invertase/react-native-firebase#4402] [Q&A] [answered]
Hi Thanks for the awesome library, just want to know the limit of data we can send using  `react-native-firebase/analytics`, I googled I found 8kb is the limit we can send in a request, so when we call ` analytics().logEvent(name, params)` it sends data to google analytics console, it's 8kb per logged action or 8kb maximum that can be sent once to google analytics?

**Comment by mikehardy** [comment]
I googled and found nothing, would be interesting to see your source. We just wrap the underlying APIs here, so this would be something to take up with either the google analytics team or the support for the underlying library of firebase-android-sdk / firebase-ios-sdk. I would be interested to hear your findings if you could report back after going upstream to the authoritative source

**Comment by whoami-shubham** [comment, marked as answer]
I referred to these sources https://developers.google.com/analytics/devguides/collection/protocol/v1/reference#using-post
https://support.google.com/analytics/thread/5867684?hl=en

**Reply by mikehardy** [reply]
Oh interesting. it's possible to batch hits so you're subdividing that 8k and I think you'd overflow the value limits on fields anyway? I had honestly never thought of 8k for an analytics hit as being any sort of a barrier but use cases may differ
